### DISCUSSION TITLE
VerifyException type should be public

### DISCUSSION THREAD

**Discussion by MJB222398** [VerifyTests/Verify#1132] [Q&A]
The VerifyException type is internal. I want to be able to to catch VerifyExceptions, currently I have to just catch all Exceptions regardless of type. Of course the constructor(s) for VerifyException can be internal but I am wondering why the type is?

**Comment by SimonCropp** [comment]
whats the scenario where u want to catch it?

**Reply by MJB222398** [reply]
@SimonCropp I don't want the exception to be thrown as I don't want my test to fail. I want to be in control of collecting and reporting failed verifications at a time of my choosing. I appreciate this is probably not how you expect people to use Verify but it feels like it would not have any negative effects were it to be public.

**Reply by SimonCropp** [reply]
> I want to be in control of collecting and reporting failed verifications at a time of my choosing. I

can you elaborate on that?

perhaps show some code that illustrates the usage

**Reply by MJB222398** [reply]
@SimonCropp sure so something like:

```
private readonly IList<string> _verificationFailures;

.......

public async Task MyTest()
{
   try
   {
       await Verify("Something")
    }
    catch (VerifyException ve)
    {
        _verificationFailures.Add(ve.Message);
    }

    .......

   try
   {
       await Verify("Something else perhaps")
    }
    catch (VerifyException ve)
    {
        _verificationFailures.Add(ve.Message);
    }

    .......

    if (__verificationFailures.Count > 0)
        throw new MySpecialException(_verificationFailures);
}
```

Something like this, apologies if I made any syntax errors, I am just typing this in here on my phone. Hopefully it makes sense?

**Reply by SimonCropp** [reply]
Have you considered using Recording https://github.com/VerifyTests/Verify/blob/main/docs/recording.md
```
[Fact]
public Task Usage()
{
    Recording.Start();
    Recording.Add("key1", "Something");
    Recording.Add("key2", "Something else perhaps");
    return Verify();
}
```

or an anon type https://github.com/VerifyTests/Verify/blob/main/docs/anonymous-types.md


    [Fact]
    public async Task Anon()
    {
        var something = "Something";
        var somethingElsePerhaps = "Something else perhaps";
        await Verify(
            new
            {
                something,
                somethingElsePerhaps,
            });
    }

**Reply by MJB222398** [reply]
@SimonCropp Recording may have the potential to support my requirements. Just tried to get it doing what I need but it creates just one file with all the recorded items in. I need a separate file for each. Also I can't seem to set the file extension? 

```
        private const string RecordingName = "XYZ";

        private readonly VerifySettings _settings;

        protected MyClass(VerifySettings settings) => _settings = settings;

        [SetUp]
        public void Setup() => Recording.Start(RecordingName);

        [TearDown]
        public async Task TearDown()
        {
            await Verifier
                .Verify(Recording.Stop(RecordingName), _settings);
        }

        // This method gets called half a dozen times or so per test typically.
        protected void VerifyThisThing(string thing, string thingName)
        {
             Recording.Add(RecordingName, thingName, thing);
        }
```
